scalalib/src/mill/scalalib/CoursierModule.scala

```diff
@@ -63,8 +68,25 @@ trait CoursierModule extends mill.Module {
    */
   def repositoriesTask: Task[Seq[Repository]] = T.task {
     import scala.concurrent.ExecutionContext.Implicits.global
+    Resolve.proxySetup()
+    val resolvedCredentials = coursier.cache.CacheDefaults.credentials.flatMap { _.get() }
     val repos = Await.result(
-      Resolve().finalRepositories.future(),
+      Resolve().finalRepositories.map {
+        _.map {
+          case x: IvyRepository =>
+            x.withAuthentication(
+              resolvedCredentials.find { c =>
+                c.matches(x.pattern.string, c.usernameOpt.getOrElse(""))
+              }.map(_.authentication)
+            )
+          case x: MavenRepository =>
+            x.withAuthentication(
+              resolvedCredentials.find { c =>
+                c.matches(x.root, c.usernameOpt.getOrElse(""))
+              }.map(_.authentication)
+            )
+        }
+      }.future(),
```

Comment:
How costly it this setup? Since it's implemented in an annonymous task, it's not cached and evaluated in every module again and again. We should consider an input target in a shared worker module to only read this configuration once.

Reply:
Maybe, we should also make sure to not override already present authentication?

Reply:
> How costly it this setup? Since it's implemented in an annonymous task, it's not cached and evaluated in every module again and again. We should consider an input target in a shared worker module to only read this configuration once.

Addressed in [910da10](https://github.com/com-lihaoyi/mill/pull/2954/commits/910da106405e63953df5fb57a7126210309cf860)

Reply:
> Maybe, we should also make sure to not override already present authentication?

Addressed in https://github.com/com-lihaoyi/mill/pull/2954/commits/504233e99b7f852ca85d327d5210e6bdd7bd7f78

Reply:
@lefou I moved the fetching of default repositories, credentials, mirrors, and proxy configuration; and the running of proxy setup to `CoursierWorkerModule`. Added links to the existing coursier implementations in doc comments, and delegated the `repositoriesTask` to the `CoursierWorkerModule`. This way, no new members are added to `CoursierModule` to interfere with currently available integrations. If we'd prefer to move them back, let me know.

The integration tests are yet `TODO`.